packages/beacon-node/src/chain/archiveStore/archiveStore.ts

```diff
@@ -123,11 +136,6 @@ export class ArchiverStore {
 
       this.prevFinalized = finalized;
 
-      await this.statesArchiverStrategy.onFinalizedCheckpoint(finalized, this.metrics);
-
-      // should be after ArchiveBlocksTask to handle restart cleanly
```

Comment:
@twoeths this comment was added like 5 years ago.... is it still relevant?  The whole unfinalized checkpoint sync brought to light some questions about startup timing and what objects are available for fork choice initialization.  This would be for a different condition but might have similar consequences so wanted to get your opinion.  We can remove the comment for future maintainers if its no biggie or can move it to somewhere more prominent if it still valid.

pinging @wemeetagain as well

Reply:
`ArchiveBlocksTask` was refactored to `archiveBlocks()` function above
I guess this comment means archiving state should be __after__  archiving block, we should keep the order and the comment, just update `ArchiveBlocksTask` to `archiveBlocks` and add more information if needed. Anyway that's not the concern of this PR.

need to make sure the order of archiving tasks is the same in your refactor @nazarhussain